ui/components/forms/flavors/compound-form/_index.scss

```diff
@@ -21,7 +21,10 @@
   }
 
   .slds-form-element {
-    width: 100%; // Default - Stretch 100% unless told otherwise using sizing helpers
+    // Default - Stretch 100% unless told otherwise using sizing helpers
+    &:not([class*="slds-size--"]) {
+      width: 100%;
+    }
```

Comment:
Very valid point. My inclination would be to reduce the specificity of this instance nested within compound forms, but that means form elements wouldn't default to 100%.

The other, more intense, option is to make the baseline utilities immutable with important tags. I know that's been suggested in another issue and I'm personally not a fan but it is an option.

I'll mull over some more ways to more intelligently reduce the specificity without losing the functionality.